### PR TITLE
Added `test_set_empty_params`

### LINKED ISSUE #6476: exp run: unexpected error - cannot represent an object: {'lr': 0.1}
# Bug Report

## Description

I have updated DVC version from 2.5.4 to latest version 2.6.4.

I am running my experiments with `dvc exp run`. I use -S option to use different parameters between experiments. Following the [get-started example](https://github.com/iterative/example-get-started/blob/master/src/prepare.py#L10) I load my params.yaml in my script and update my model config programatically. 

If my `params.yaml` is emtpy and I run `dvc exp run -S optimizer.lr=0.1`:

- `dvc 2.5.4`: experiment runs without problems
- `dvc 2.6.4`: experiment crashes returning a confusing error:` ERROR: unexpected error - cannot represent an object: {'lr': 0.1}`

### Reproduce

1. dvc init
2. Copy dataset.zip to the directory
3. dvc add dataset.zip
4. create empty params.yaml
5. dvc exp run -S optimizer.lr=0.1

-->

### Expected

Experiment runs without returning any error.

### Environment information

**Output of `dvc doctor`:**

```console
$ dvc doctor
DVC version: 2.6.4 (pip)
---------------------------------
Platform: Python 3.8.10 on Linux-4.15.0-96-generic-x86_64-with-glibc2.10
Supports:
        gdrive (pydrive2 = 1.9.1),
        http (requests = 2.26.0),
        https (requests = 2.26.0)
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/sdb1
Caches: local
Remotes: None
Workspace directory: ext4 on /dev/sdb1
Repo: dvc, git
```

**Additional Information (if any):**

I wouldn't like to have to set my `params.yaml` manually. This would force to replicate my config twice. Most DL frameworks have config files where model description and training hyperparameters are stored. So I would like to update those values simply using the -S flag.

### PR DESCRIPTION
**Pull request by daavoo**
* [X] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [X] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

- Added explicit test for https://github.com/iterative/dvc/issues/5477
- Added regression test for  #6476

### CODE REVIEW

File: dvc/utils/serialize/_yaml.py [hunk 1]
```diff
@@ -50,6 +51,8 @@ def _get_yaml():
     # tell Dumper to represent OrderedDict as normal dict
     yaml_repr_cls = yaml.Representer
     yaml_repr_cls.add_representer(OrderedDict, yaml_repr_cls.represent_dict)
+    yaml_repr_cls.add_representer(benedict, yaml_repr_cls.represent_dict)
```
Comment (skshetry) [inline dvc/utils/serialize/_yaml.py:54]: benedict should never escape when we merge params from `--set-params`. This may result in loss of commits. 
A fix might be to merge the params in the file recursively from the `--set-params` benedict dictionary.

Comment (daavoo) [reply dvc/utils/serialize/_yaml.py:54]: Not sure I'm fully understanding, what do you mean by `escape` ?

Comment (daavoo) [reply dvc/utils/serialize/_yaml.py]: Removed that and replaced with just a regression test currently checking that a exception is being raised. I guess that if we go with https://github.com/iterative/dvc/issues/6423#issuecomment-908028959 we should make that test pass

Comment (skshetry) [reply dvc/utils/serialize/_yaml.py]: #6476 is more urgent than #6423, though we could do it together too.

File: dvc/utils/serialize/_yaml.py [hunk 2]
```diff
@@ -50,6 +50,7 @@ def _get_yaml():
     # tell Dumper to represent OrderedDict as normal dict
     yaml_repr_cls = yaml.Representer
     yaml_repr_cls.add_representer(OrderedDict, yaml_repr_cls.represent_dict)
+
```
Comment (daavoo) [inline dvc/utils/serialize/_yaml.py:53]:
```suggestion
```

File: dvc/utils/serialize/_yaml.py (hunk 1 above)
Comment (daavoo) [reply dvc/utils/serialize/_yaml.py]: I'm not sure but maybe we could add this tests for now and discuss in https://github.com/iterative/dvc/pull/6521 how we want to handle #5477 (as it would make #6476 no longer a bug but a different exception)

**Comment by skshetry** [comment]
Sorry for the late review. I have incorporated this test in https://github.com/iterative/dvc/pull/6706#discussion_r718207470, as it gets automatically fixed after updating benedict to the latest version. This bug happened because we were handling an empty dictionary in a different way because of a bug in benedict which we don't need to handle now at all.
